docs/errors/E0720.md

```diff
@@ -0,0 +1,39 @@
+# E0720: calling `let` on an assignment is ambiguous
+
+If `let` is declared as a function, then this code:
+
+```javascript
+function let(arg) {
+  console.log(arg);
+}
+
+const y = {x: "hello", z: "world"};
+let ({x} = y);
+```
+
+is actually equivalent to:
+
+```javascript
+function let(arg) {
+  console.log(arg);
+}
+
+const y = {x: "hello", z: "world"};
+let(y);
+```
+
+instead of destructuring `y`.
```

Comment:
Must fix: This explanation is wrong. You removed the assignment to `x`. The second snippet should look more like this:
```javascript
function let(arg) {
  console.log(arg);
}

const y = {x: "hello", z: "world"};
({x} = y);      // <-- your explanation is missing this line.
let(y);
```
The explanation also says "If `let` is declared as a function", but it doesn't matter what `let` is declared as (or if it's even declared at all).

Also, the style of the explanation doesn't match our typical style. We shouldn't split the sentence into pieces with code blocks in the middle, and we shouldn't use two code blocks to explain the buggy code.

I suggest a rewrite of this section to something like the following:
```suggestion
In JavaScript, variables can be named `let`. If a `let` statement has parentheses in the wrong spot, `let` itself is treated as a variable name. The statement is thus a function call, not a normal `let` statement:

```javascript
const person = {name: "Alice", age: 9001};
// The following line does
// *not* declare any variables.
let ({name, age} = person);
```-
```